### DISCUSSION TITLE
dnscrypt-proxy in pi hole server

### DISCUSSION THREAD

**Discussion by jmssil** [DNSCrypt/dnscrypt-proxy#2666] [Q&A]
How do I make dnscrypt-proxy listen on port 5353 instead of 53? I need that to make pi hole, which is running on the same machine, use dnscrypt as its DNS server: https://discourse.pi-hole.net/t/pihole-with-dnscrypt-proxy/71325/3

**Comment by jedisct1** [comment]
https://github.com/DNSCrypt/dnscrypt-proxy/blob/de16f22c0cea8606d0a270481633824f175e38a3/dnscrypt-proxy/example-dnscrypt-proxy.toml#L35-L42

**Reply by jmssil** [reply]
I tried that but I think it does not work. Pihole complains about:

`DNSMASQ_CONFIG	FTL failed to start due to failed to create listening socket for port 53: Address already in use
`
Seems like dnscrypt-proxy is still using those ports:

```
 [2024-07-30 13:30:49] [NOTICE] Wiring systemd TCP socket #0, dnscrypt-proxy.socket, 127.0.2.1:53
 [2024-07-30 13:30:49] [NOTICE] Wiring systemd UDP socket #1, dnscrypt-proxy.socket, 127.0.2.1:53
 [2024-07-30 13:30:49] [NOTICE] Wiring systemd TCP socket #2, dnscrypt-proxy.socket, 127.0.0.1:5353
 [2024-07-30 13:30:49] [NOTICE] Wiring systemd UDP socket #3, dnscrypt-proxy.socket, 127.0.0.1:5353
```

```
# cat /etc/dnscrypt-proxy/dnscrypt-proxy.toml
# Empty listen_addresses to use systemd socket activation
listen_addresses = ['127.0.0.1:2053']
server_names = ['cloudflare']
```

**Comment by lifenjoiner** [comment]
Following the document you will get https://github.com/DNSCrypt/dnscrypt-proxy/wiki/Installation-on-Pi-hole

**Reply by jmssil** [reply]
Thanks, it worked. I think that dnscrypt-proxy from Debian sid additionally installs dnsmasq or some other software that listens on port 53 and therefore pihole can't run.
